gossipd/gossip.c

```diff
@@ -1519,6 +1521,27 @@ static void setup_listeners(struct daemon *daemon, u16 portnum)
 		status_failed(STATUS_FAIL_INTERNAL_ERROR,
 			      "Could not bind to a network address on port %u",
 			      portnum);
+
+	/* Optional UNIX socket */
+	if (localsocket) {
```

Comment:
This comes from the `fromwire_gossipctl_init`.  In `towire_gossipctl_init` a `NULL` pointer is converted to a 0-length array on-the-wire, I am uncertain if a 0-length array on-the-wire will be converted by `fromwire_gossipctl_init` to a `NULL` pointer.  I  shall check later when I can get access to my hacking node.

Reply:
No issue here: turns out 0-length arrays on-the-wire are converted to `NULL` by `fromwire_*` functions, so ok.